include/ensmallen_bits/cmaes/transformation_policies/boundary_box_constraint.hpp

```diff
@@ -0,0 +1,166 @@
+/**
+ * @file boundary_box_constraint.hpp
+ * @author Suvarsha Chennareddy
+ *
+ * Boundary Box Transformation.
+ *
+ *
+ * ensmallen is free software; you may redistribute it and/or modify it under
+ * the terms of the 3-clause BSD license.  You should have received a copy of
+ * the 3-clause BSD license along with ensmallen.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+namespace ens {
+  /**
+  * More often than not, coordinates must be bounded by some constraints.
+  * In a particular case, the domain of a specific function is restricted 
+  * by boundaries.
+  * The implemented transformation transforms given coordinates into a region 
+  * bounded by the given lower and upper bounds (a box). First, the 
+  * coordinates are shifted into a feasible preimage bounded by lower_bound - al 
+  * and upper_bound + au where al and au and calculated internally. 
+  * These shifted coordinates are then transformed into coordinates bounded by 
+  * lower_bound and upper_bound. It is an identity transformation in between 
+  * the lower and upper bounds.
+  * The inverse transformation transforms the coordinates back to the preimage.
```

Comment:
The inverse transformation is never used in these changes---do you intend to use it, or should we remove the implementation?  (Not a great idea to keep unused code)

Reply:
Yeah...I was a bit hesitant on adding it, we really don't need it at all in this case. Is commenting out fine?

Reply:
I would just remove it.  If it's left in but commented out, my experience is that nobody will ever come back and uncomment it; it'll just be dead code.